tests/testthat/test_Plot_synergy.R

```diff
@@ -0,0 +1,214 @@
+      0, 10.192587, 14.67011, 27.47910, 12.08755, 5.526368,
+      0,  5.586906, 11.69542, 34.82529, 24.55566, 9.158482,
+      0,  8.458418, 14.96854, 36.53241, 25.40108, 9.456784,
+      0,  8.772475, 15.88465, 36.98242, 26.60314, 9.690151,
+      0,  8.536783, 18.14888, 37.28297, 28.90676, 9.654226),
+    nrow=6, ncol=6, byrow=TRUE)
+
+  dimnames(scores) <- list(
+    .GetResponseAndScoreRowNames(),
+    .GetResponseAndScoreColumnNames()
+  )
+
+  return(scores)
+}
+
+.GetScores2 <- function() {
+  scores <- matrix(
+    c(0,   0.000000,   0.00000,   0.00000,   0.00000,   0.000000,
+      0,  -9.147316, -16.23292, -19.87011,  -3.06092,   5.994880,
+      0, -15.630544, -23.30691, -29.43377, -10.30092,   6.000334,
+      0, -11.452155, -19.23089, -26.34151, -28.46625,  -6.057822,
+      0, -14.793930, -22.50455, -29.86705, -33.87227, -14.592597,
+      0, -24.888500, -21.73267, -18.37694, -14.59779,  -6.714009),
+    nrow=6, ncol=6, byrow=TRUE)
+
+  dimnames(scores) <- list(
+    .GetResponseAndScoreRowNames(),
+    .GetResponseAndScoreColumnNames()
+  )
+
+  return(scores)
+}
+
+.GetDrugPairs <- function() {
+  return(
+    data.frame(
+      'drug.row' = c('ispinesib', 'canertinib'),
+      'drug.col' = c('ibrutinib', 'ibrutinib'),
+      'concUnit' = c(       'nM',        'nM'),
+      'blockIDs' = c(          1,           2)))
+}
+
+test_that("compute_title_works", {
+  expect_equal(18.04229, mean(.GetScores1()[-1,-1]), tolerance=1e-05)
+  expect_equal("Average synergy: 18.042 (ZIP)",
+               .ComputePlotTitle(.GetScores1(), NULL, NULL, 'ZIP'))
+
+  expect_equal(-16.33909, mean(.GetScores2()[-1,-1]), tolerance=1e-05)
+  expect_equal("Average synergy: -16.339 (ZIP)",
+               .ComputePlotTitle(.GetScores2(), NULL, NULL, 'ZIP'))
+})
+
+test_that("round_up_to_ten_works", {
+  expect_equal(10, .RoundUpToNextTen(1.5))
+  expect_equal(10, .RoundUpToNextTen(-1.5))
+  expect_equal(10, .RoundUpToNextTen(5))
+  expect_equal(10, .RoundUpToNextTen(5.1))
+  expect_equal(10, .RoundUpToNextTen(-9.9))
+  expect_equal(20, .RoundUpToNextTen(10))
+  expect_equal(40, .RoundUpToNextTen(.GetScores1()))
+  expect_equal(40, .RoundUpToNextTen(.GetScores2()))
+})
+
+test_that("get_plot_data_works", {
+  input_data <- list(
+    dose.response.mats = list(.GetResponses1(), .GetResponses2()),
+    scores = list(.GetScores1(), .GetScores2()),
+    drug.pairs = .GetDrugPairs(),
+    method = "ZIP"
+  )
+
+  plot_data1 <- .GetPlotData(input_data, 1, NULL, NULL, NULL, NULL, 3)
+  plot_data2 <- .GetPlotData(input_data, 2, NULL, NULL, NULL, NULL, 3)
+
+  expect_equal("Average synergy: 18.042 (ZIP)", plot_data1$data$plot.title)
+  expect_equal("Average synergy: -16.339 (ZIP)", plot_data2$data$plot.title)
+  expect_equal("ispinesib.ibrutinib", plot_data1$metadata$file.name.prefix)
+  expect_equal("canertinib.ibrutinib", plot_data2$metadata$file.name.prefix)
+  # TODO: make some kind of assertions about kriging output in mat.tmp
```

Comment:
It might be easier to provide the kriging ouput with len = 1.

Reply:
I think it will be easier still when you have patched up .ExtendedScores(). :)  I would like to let this TODO remain for future work, rather than taking care of it immediately.  This change is already quite large.